File: Mobile/Mobile Beta/app/src/main/java/Endereco.java
public class Endereco {
    public String CEP;
    public String Pais;
    public String Estado;
    public String Cidade;
    public String Bairro;
    public String Logradouro;
    public int Numero;
    public String Complemento;
    public int IdCliente;
    public Endereco(String cep, String pais, String estado, String cidade,String bairro , String logradouro, int numero, String complemento, int idCli){

        this.CEP = cep;
        this.Pais = pais;
        this.Estado = estado;
        this.Cidade = cidade;
        this.Bairro = bairro;
        this.Logradouro = logradouro;
        this.Numero = numero;
        this.Complemento = complemento;
        this.IdCliente = idCli;



    }

    public Endereco(){
        this.CEP = null;
        this.Pais = null;
        this.Estado = null;
        this.Cidade = null;
        this.Bairro = null;
        this.Logradouro = null;
        this.Numero = 0;
        this.Complemento = null;
        this.IdCliente = 0;
    }



}

File: Aplicação/Scrolls/Scrolls.Entities/Endereco.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Scrolls.Entities
{
    public class Endereco
    {
        public int Id { get; set; }

        [Required, RegularExpression(@"^\d{5}(\-)(\d{3})?$")]
        public string CEP { get; set; }
        
        [Required]
        public string Pais { get; set; }

        [Required]
        public string Estado { get; set; }

        [Required]
        public string Cidade { get; set; }

        [Required]
        public string Bairro { get; set; }

        [Required]
        public string Logradouro { get; set; }
        
        [MaxLength(100, ErrorMessage ="O campo deve ter apenas 100 caracteres.")]
        public string Complemento { get; set; }

        [Required(ErrorMessage = "O campo número não pode estar vazio.")]
        public int Numero { get; set; }

        public virtual Cliente Cliente { get; set; }
        public int? ClienteId { get; set; }

        public virtual IList<Venda> Vendas { get; set; }
    }
}
